shell/core/plugin.ts

```diff
@@ -132,6 +132,11 @@ export class Plugin implements IPlugin {
         console.warn(`Layouts have been deprecated. We still have parent routes which use the same name and styling as the previous layouts. You should specify a parent, we're currently setting the parent to 'default'`); // eslint-disable-line no-console
         parentOverride = 'default';
       }
+
+      // Fix for Home page components with wrong layout - need to ensure the parentOverride is set
+      if (typelessRoute.component && typelessRoute.name === 'home' && typelessRoute.path === '/home') {
+        parentOverride = 'home_layout';
```

Comment:
The warning above mentions we're setting the override to default, but then here we're silently setting to home_layout.

Should this be in an if/else above with a similar but more specific warning log?

Reply:
@richard-cox The home page component we use in our example uses the 'plain' layout - but this is NOT the parent for the home page route, so that is why the mechanism is broken.

The layout for the home page route does not have a name - possibly there is a work around using the empty parent route that I will check - but adding an explicit name of 'home_layout' makes sense.

I kept the change simple - because the home page is in the 'home_layout', the parent must be 'home_layout' in order to replace it.

For 2.10, we could take a more breaking change approach by adding the layout name 'home_layout' and then updating the 'setHomeComponent' or whatever it is to work correctly - that will break extensions that use the `addRoute` method though.

TL;DR - I personally think this is okay - @codyrancher can weigh in

Reply:
I can't think of a less invasive fix so I'm fine with it.

Reply:
the approach is good! just wondered if we could avoid the misleading console message.

my suggestion was bad, adding the condition in the middle of the others work message wise but breaks functionality